retrieval/scrape.go

```diff
@@ -496,10 +496,10 @@ func (sl *scrapeLoop) report(start time.Time, duration time.Duration, err error)
 		Value:     model.SampleValue(duration.Seconds()),
 	}
 
-	if err := sl.reportAppender.Append(healthSample); err != nil {
+	if err := sl.reportAppender.Append(context.TODO(), healthSample); err != nil {
 		log.With("sample", healthSample).With("error", err).Warn("Scrape health sample discarded")
 	}
-	if err := sl.reportAppender.Append(durationSample); err != nil {
+	if err := sl.reportAppender.Append(context.TODO(), durationSample); err != nil {
```

Comment:
Yes, changing it from TODO to Background doesn't really help. In the end usages of context should probably be connected to `main()` or one level below.

If there's no use for anyone right now, let's leave it out for now. I like contexts, but they also add complexity. As you said, we never really want to do cancelation and I cannot see us passing in any values via it.